File: Assets/MLRepo/ml.72201f1e-ba18-487e-adc4-4381ae10e21b.davidmatos_nonphysicsdoor/NonPhysicsDoor/Scripts/NonPhysicsDoorHandle.lua
Namespace("davidmatos_nonphysicsdoor");
do -- script NonPhysicsDoorHandle 
	
	-- get reference to the script
	local NonPhysicsDoorHandle = LUA.script;
	
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||

	local gameObject_originalDoorHandlePosition = SerializedField("Original Door Handle Position", GameObject);
	local number_forceReleaseDistance = SerializedField("Force Release Distance", Number);

	--|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||

	local mlplayer_localPlayer = nil;
	local mlgrab_grab = nil;

	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
	--|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||

	-- start only called at beginning
	function NonPhysicsDoorHandle.Start()
		mlplayer_localPlayer = Room.GetLocalPlayer();

		mlgrab_grab = NonPhysicsDoorHandle.gameObject.GetComponent(MLGrab);
	end

	-- update called every frame
	function NonPhysicsDoorHandle.Update()
		local distanceFromDoor = Vector3.Distance(NonPhysicsDoorHandle.gameObject.transform.position, gameObject_originalDoorHandlePosition.transform.position);

		if(distanceFromDoor > number_forceReleaseDistance) then
			mlgrab_grab.ForceRelease();
		end

		if(mlgrab_grab.CurrentUser == nil) then
			NonPhysicsDoorHandle.gameObject.transform.position = gameObject_originalDoorHandlePosition.transform.position;
			NonPhysicsDoorHandle.gameObject.transform.localRotation = Quaternion(0, 0, 0, 0);
		end
	end
end

File: Assets/MLRepo/ml.72201f1e-ba18-487e-adc4-4381ae10e21b.davidmatos_nonphysicsdoor/NonPhysicsDoor/Scripts/NonPhysicsDoorHandle.cs
using ML.SDK;
using UnityEngine;

    public class NonPhysicsDoorHandle : MonoBehaviour
    {
        //|||||||||||||||||||||||||||||||||||||||||||||| PUBLIC VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
        [Header("Door Handle Settings")]
        [SerializeField] private GameObject originalDoorHandlePosition;
        [SerializeField] private float forceReleaseDistance = 0.5f;

        //|||||||||||||||||||||||||||||||||||||||||||||| PRIVATE VARIABLES ||||||||||||||||||||||||||||||||||||||||||||||
        private MLPlayer localPlayer;
        private MLGrab grabComponent;

        //|||||||||||||||||||||||||||||||||||||||||||||| UNITY FUNCTIONS ||||||||||||||||||||||||||||||||||||||||||||||
        private void Start()
        {
            localPlayer = MassiveLoopRoom.GetLocalPlayer();
            grabComponent = GetComponent(typeof(MLGrab)) as MLGrab;

            if (originalDoorHandlePosition == null)
            {
                Debug.LogError("Original Door Handle Position reference is missing!", this);
            }
        }

        private void Update()
        {
            if (originalDoorHandlePosition == null || grabComponent == null) return;

            // Check distance from original position
            float distanceFromDoor = Vector3.Distance(
                transform.position,
                originalDoorHandlePosition.transform.position);

            // Force release if moved too far
            if (distanceFromDoor > forceReleaseDistance)
            {
                grabComponent.ForceRelease();
            }

            // Reset position when not grabbed
            if (grabComponent.CurrentUser == null)
            {
                transform.position = originalDoorHandlePosition.transform.position;
                transform.localRotation = Quaternion.identity;
            }
        }
    }